pkg/scyllaclient/client_agent.go

```diff
@@ -254,6 +254,29 @@ func (ni *NodeInfo) SupportsSafeDescribeSchemaWithInternals() (SafeDescribeMetho
 	return "", nil
 }
 
+// SupportsScyllaBackupRestoreAPI returns whether node exposes backup/restore API
+// that can be used instead of the Rclone API for backup/restore tasks.
+func (ni *NodeInfo) SupportsScyllaBackupRestoreAPI() (bool, error) {
+	// Check master builds
+	if scyllaversion.MasterVersion(ni.ScyllaVersion) {
+		return true, nil
+	}
+	// Check OSS
+	supports, err := scyllaversion.CheckConstraint(ni.ScyllaVersion, ">= 6.3, < 2000")
+	if err != nil {
+		return false, errors.Errorf("Unsupported Scylla version: %s", ni.ScyllaVersion)
+	}
+	if supports {
+		return true, nil
+	}
+	// Check ENT
+	supports, err = scyllaversion.CheckConstraint(ni.ScyllaVersion, ">= 2024.3")
+	if err != nil {
+		return false, errors.Errorf("Unsupported Scylla version: %s", ni.ScyllaVersion)
+	}
+	return supports, nil
+}
+
```

Comment:
So, it's basing on the release version. 
It's confirmed that Scylla exposes wtih appi beginning with 6.3 and 2024.3 ? 
Is there any other possibility of checking the availablity of this endpoint ? Maybe simple HEAD request to the endpoint ?

Reply:
It's not confirmed, we would need  to come back to this part after an actual Scylla release supports native backup/restore API.

Reply:
Can we assume that if endpoint is available then it means scylla supports it ? If so, then I suggest to probe api with HEAD request.

Reply:
In general I like the idea of not relying on version checks, but proposed approach requires Scylla to handle HEAD requests and I don't think that's the case. At least I can't see them in the swagger definitions or swagger UI.

Reply:
OK, they don't support HEAD requests, at least this is what I see trying to `curl` for it from localhost.
GET responds OK, HEAD gives 404 (not found).

Then maybe, there is some configuration parameter available to GET which can answer whether the feature is available ?

Reply:
Not sure about this suggestion. Wouldn't it be difficult to distinguish potential errors coming from API not being exposed and connectivity issues?
Also, the initial idea was to check Scylla swagger definitions, but it's currently not possible due to https://github.com/scylladb/scylladb/issues/16424.

Reply:
If you have connectivity issue, then you got network level error. You won't get response with HTTP code.

Reply:
> Then maybe, there is some configuration parameter available to GET which can answer whether the feature is available ?

Could you elaborate?

Reply:
If core uses some configuration parameter related to the backup API in scylla that is exposed together with this API, there could be simple GET performed on this config param.
404 code means that the API is not available.

But, if the backup API means that there are only POST methods exposed that are changing the state then I think you need to stay with the current approach where you basically checks the versions number.